spec.html

```diff
@@ -734,6 +745,283 @@ <h1>Example Source Text Module Record Graphs</h1>
+            <th>[[AsyncEvaluating]]</th>
+            <th>[[AsyncParentModules]]</th>
+            <th>[[PendingAsyncDependencies]]</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <th>_A_</th>
+            <td>0</td>
+            <td>0</td>
+            <td>*true*</td>
+            <td>&laquo; &raquo;</td>
+            <td>2 (_B_ and _C_)</td>
+          </tr>
+          <tr>
+            <th>_B_</th>
+            <td>1</td>
+            <td>0</td>
+            <td>*true*</td>
+            <td>&laquo; _A_ &raquo;</td>
+            <td>1 (_D_)</td>
+          </tr>
+          <tr>
+            <th>_C_</th>
+            <td>2</td>
+            <td>0</td>
+            <td>*true*</td>
+            <td>&laquo; _A_ &raquo;</td>
+            <td>2 (_D_ and _E_)</td>
+          </tr>
+          <tr>
+            <th>_D_</th>
+            <td>3</td>
+            <td>0</td>
+            <td>*true*</td>
+            <td>&laquo; _B_, _C_ &raquo;</td>
+            <td>0</td>
+          </tr>
+          <tr>
+            <th>_E_</th>
+            <td>4</td>
+            <td>4</td>
+            <td>*true*</td>
+            <td>&laquo; _C_ &raquo;</td>
+            <td>0</td>
+          </tr>
+        </tbody>
+      </table>
+    </emu-table>
+
+    <p>
+      Let us assume that _E_ finishes executing first.
+      When that happens, AsyncModuleExecutionFulfilled is called, _E_.[[AsyncEvaluating]] is set to *false* and _C_.[[PendingAsyncDependencies]] is decremented to become 1.
+      The fields of the updated modules are as given in <emu-xref href="#table-module-graph-cycle-async-fields-2"></emu-xref>.
+    </p>
+
+    <emu-table id="table-module-graph-cycle-async-fields-2" caption="Module fields after module _E_ finishes executing">
+      <table>
+        <thead>
+          <tr>
+            <th>Module</th>
+            <th>[[DFSIndex]]</th>
+            <th>[[DFSAncestorIndex]]</th>
+            <th>[[AsyncEvaluating]]</th>
+            <th>[[AsyncParentModules]]</th>
+            <th>[[PendingAsyncDependencies]]</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <th>_C_</th>
+            <td>2</td>
+            <td>0</td>
+            <td>*true*</td>
+            <td>&laquo; _A_ &raquo;</td>
+            <td>1 (_D_)</td>
+          </tr>
+          <tr>
+            <th>_E_</th>
+            <td>4</td>
+            <td>4</td>
+            <td>*false*</td>
+            <td>&laquo; _C_ &raquo;</td>
+            <td>0</td>
+          </tr>
+        </tbody>
+      </table>
+    </emu-table>
+
+    <p>
+      _D_ is next to finish (as it was the only module that was still executing).
+      When that happens, AsyncModuleExecutionFulfilled is called again and _D_.[[AsyncEvaluating]] is set to *false*.
+      Both _B_.[[PendingAsyncDependencies]] and _C_.[[PendingAsyncDependencies]] are decremented to become 0, so ExecuteAsyncModule is called and they start executing.
```

Comment:
Can we rather be explicit here that _first_ _B_.[[PendingAsyncDependencies]] is decremented to become 0, resulting in execution of B. Then after B has fully finished executing (and possibly even any synchronous parents), then _C_.[[PendingAsyncDependencies]] is decremented.